File: packages/rook_sdk_health_connect/lib/src/data/mapper/rook_configuration_mappers.dart
import 'package:rook_sdk_core/rook_sdk_core.dart';
import 'package:rook_sdk_health_connect/src/data/mapper/rook_environment_mappers.dart';
import 'package:rook_sdk_health_connect/src/data/proto/protos.pb.dart';

extension RookConfigurationMappers on RookConfiguration {
  RookConfigurationProto toProto() {
    return RookConfigurationProto(
      clientUUID: clientUUID,
      secretKey: secretKey,
      environment: environment.toProto(),
      enableBackgroundSync: enableBackgroundSync,
    );
  }
}

File: packages/rook_sdk_health_connect/android/src/main/kotlin/com/rookmotion/rook_sdk_health_connect/mapper/RookConfigurationMappers.kt
package com.rookmotion.rook_sdk_health_connect.mapper

import com.rookmotion.rook.sdk.domain.model.RookConfiguration
import com.rookmotion.rook_sdk_health_connect.data.proto.RookConfigurationProto

fun RookConfigurationProto.toRookConfiguration(): RookConfiguration {
    return RookConfiguration(
        clientUUID,
        secretKey,
        environment.toRookEnvironment(),
    )
}
